### PR TITLE
Add tests for XML writer

### PR DESCRIPTION
**Pull request by dstansby**
This should fix https://github.com/brainglobe/brainglobe-napari-io/issues/10 (the XML writers not working), and adds associated tests.

Note that the XML "multiple file" writer works fine for writing a single file to (as far as I can tell from local testing), so to reduce code duplication and the number of options in the plugin menu I've removed the "single file" writer.

### CODE REVIEW

**Comment by adamltyson** [comment]
This has fixed saving of a single layer to xml, but the multiple layer writing (e.g. >1 points layer to a single XML file) still seems to be broken for me.

**Comment by dstansby** [comment]
Thanks for testing. I've refactored the code a bit to reduce duplication, and it seems to be working for me locally. @adamltyson could you check again? The tests failing are unrelated, and fixed in https://github.com/brainglobe/brainglobe-napari-io/pull/15

**Comment by adamltyson** [comment]
Maybe I mistyped on my last comment, because now I can save multiple layers, but not a single layer. I get:
```
No data written! There may be no plugins capable of writing these 1 layers to /home/adam/Desktop/test_single_layer.xml.
```

**Comment by dstansby** [comment]
I can reproduce that. The multiple writer seems capable of saving data from a single layer though (at least from local tests), so I'd propose that we remove the single layer writer and have just one points writer that can hangle both single and multiple layers?

**Review by paddyroddy — APPROVED** [review]

File: brainglobe_napari_io/tests/test_cellfinder_writer.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+def test_xml_roundrip(tmpdir):
+    # Check that a read in XML file can also be written back out
+    xml_file = test_data_dir / "cell_classification.xml"
+    layers = reader_xml.xml_reader(xml_file)
+    assert len(layers) == 2
+
+    paths = writer_xml.write_multiple_points_to_xml(
+        str(tmpdir / "multiple.xml"), layers
+    )
```
[the remark this answers is not in the document]
Comment (adamltyson) [reply brainglobe_napari_io/tests/test_cellfinder_writer.py:21]: I guess it should be able to. The file will [eventually just be opened with `open`](https://github.com/adamltyson/imlib/blob/625193be4a586d9040a48df9d51dbdd3a17c7d06/imlib/IO/cells.py#L130).

**Pull request merged**